### ISSUE TITLE
VisualRust doesn't play well with Rust 1.7, the required registry entry isn't set

### ISSUE THREAD

**Issue by marcinmrotek** [PistonDevelopers/VisualRust#207]
VisualRust can't find the Rust 1.7 compiler after a fresh install of both on Windows 10 with Visual Studio 2015. I've fixed it by adding a registry entry mimicking the one mentioned in #187. There was no "Mozilla Foundation" key, so I've added the whole tree "Mozilla Foundation\Rust stable (64-bit)\1.7" and set "InstallDir" to the root of the installation directory (not the /bin).

![rust_registry](https://cloud.githubusercontent.com/assets/8394815/13513531/c04f8352-e1a1-11e5-83a9-14b03d97308a.png)

Before I did that, I tried reinstalling and repairing the compiler, and it didn't set the registry entry, so I presume it just stopped doing this in the newer versions?

**Comment by retep998** [comment]
VisualRust really shouldn't rely on a registry key anyway, since some people (like me) install Rust using multirust-rs.

**Comment by mihe** [comment]
Can confirm that I ran into this issue as well, and solved it using the same method as @marcinmrotek. If nothing else, couldn't looking for rustc.exe be a potential fallback, considering it'll most likely be in `%PATH%`?

**Comment by retep998** [comment]
I'd prefer if this supported multirust-rs, as in it could be configured to set MULTIRUST_TOOLCHAIN to the version I want, and depend on multirust-rs being in the path.

**Comment by vosen** [comment]
We support multirust-rs, although the discovery order is local official installation ---> multirust ---> global official installation (multirust and global path used to be switched around but I've just changed it). If we don't pick a multirust installation or pick wrong path/version it's a bug.
As for looking in `%PATH%`, I've got a mixed feeling about this. Usually, language plugins (not only in Visual Studio) rely on proper paths in the registry. If there's no official distribution, then you usually get an option to fiddle with the paths yourself (and we want to enable this with #172). `%PATH%` is left for you to safely break your command line in any way you want, knowing that this won't break your IDE.
